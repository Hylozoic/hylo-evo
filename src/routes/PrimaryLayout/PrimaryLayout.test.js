import { MemoryRouter, Redirect } from 'react-router'
import React from 'react'
import { shallow } from 'enzyme'

import orm from 'store/models'
import
PrimaryLayout,
{
  redirectIfCommunity,
  RedirectToCommunity,
  RedirectToCreateCommunityFlow
} from './PrimaryLayout'

it('shows NotFound if a currentUser is loaded and the community does not exist', () => {
  const wrapper = shallow(<PrimaryLayout isCommunityRoute currentUser={{}} />)
  expect(wrapper).toMatchSnapshot()
})

it('shows nothing for a community route if the community and currentUser are not loaded', () => {
  const wrapper = shallow(<PrimaryLayout isCommunityRoute />)
  expect(wrapper.name()).toEqual('Loading')
})

it('shows normal children for a community route if the community is loaded', () => {
  const wrapper = shallow(<PrimaryLayout
    isCommunityRoute
    location={{pathname: '/'}}
    community={{id: '1'}}
    currentUser={{}} />)
  expect(wrapper.name()).toEqual('div')
})

describe('RedirectToCommunity', () => {
  let session

  beforeEach(() => {
    session = orm.mutableSession(orm.getEmptyState())
    session.Me.create({ id: '1' })
  })

  it('shows a Loading component if currentUser not set yet', () => {
    const wrapper = shallow(<RedirectToCommunity currentUser={null} />)
    const renderWrapper = shallow(wrapper.props().render())
    expect(renderWrapper.find('[data-styleName="loading-top"]').length).toBe(1)
  })

  it('shows a Loading component if currentUser has no memberships', () => {
    const me = session.Me.first()
    const wrapper = shallow(redirectIfCommunity(me)())
    expect(wrapper.find('[data-styleName="loading-top"]').length).toBe(1)
  })

  it('sets `to` prop of Redirect correctly', () => {
    session.Community.create({ id: '1', slug: 'foo' })
    session.Membership.create({ id: '1', community: '1' })
    const me = session.Me.first()
    me.set('memberships', ['1'])

    const wrapper = shallow(<MemoryRouter>
      {redirectIfCommunity(me)()}
    </MemoryRouter>)

    const expected = '/c/foo'
    const actual = wrapper.find(Redirect).props().to
    expect(actual).toBe(expected)
  })
})

describe('RedirectToCreateCommunityFlow', () => {
  it('returns null if a signup is in progress', () => {
    const currentUser = {settings: {signupInProgress: true}}
    const wrapper = shallow(<RedirectToCreateCommunityFlow currentUser={currentUser} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('returns null if there are memberships', () => {
    const hasMemberships = true
    const wrapper = shallow(<RedirectToCreateCommunityFlow hasMemberships={hasMemberships} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('returns null if pathname starts with /signup', () => {
    const pathname = '/signup/any-path-here'
    const wrapper = shallow(<RedirectToCreateCommunityFlow pathname={pathname} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('returns null if pathname starts with /create-community', () => {
    const pathname = '/create-community/any-path-here'
    const wrapper = shallow(<RedirectToCreateCommunityFlow pathname={pathname} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('returns null if returnToURL is /h/use-invitation', () => {
    const currentUser = {settings: {signupInProgress: false}}
    const pathname = '/'
    const returnToURL = '/h/use-invitation'
    const wrapper = shallow(<RedirectToCreateCommunityFlow
      pathname={pathname}
      returnToURL={returnToURL}
      currentUser={currentUser}
    />)
    expect(wrapper).toMatchSnapshot()
  })
  it('returns a redirect from create community if: signup is complete, a user does not have memberships, and is not already on /signup or /create-community', () => {
    const currentUser = {settings: {signupInProgress: false}}
    const hasMemberships = false
    const pathname = '/'
    const wrapper = shallow(<RedirectToCreateCommunityFlow
      pathname={pathname}
      hasMemberships={hasMemberships}
      currentUser={currentUser}
    />)
    const expected = '/create-community/name'
    const actual = wrapper.find(Redirect).props().to
    expect(actual).toBe(expected)
    expect(wrapper).toMatchSnapshot()
  })
})
