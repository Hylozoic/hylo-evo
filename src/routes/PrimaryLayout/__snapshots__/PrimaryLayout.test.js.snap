// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows DownloadAppModal if the user is on mobile 1`] = `
<div
  data-stylename="container"
>
  <Connect(Drawer)
    data-stylename="drawer hidden"
  />
  <Connect(TopNav)
    data-stylename="top"
    onClick={[Function]}
  />
  <div
    data-stylename="main"
    onClick={[Function]}
  >
    <withRouter(Connect(Navigation))
      collapsed={false}
      data-stylename="left"
      showTopics={true}
    />
    <div
      data-stylename="center"
      id="center-column"
    >
      <RedirectToSignupFlow
        pathname="path"
      />
      <RedirectToCommunity
        path="/"
      />
      <RedirectToCommunity
        path="/app"
      />
      <Switch>
        <Route
          exact={true}
          path="/c/:slug/tag/:topicName"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/c/:slug/join/:accessCode/tag/:topicName"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/p/:postId"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/u/:id"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/c/:slug/about"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/c/:slug/people"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/c/:slug/invite"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/c/:slug/events"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/c/:slug/projects"
          render={[Function]}
        />
        <Route
          component={[Function]}
          exact={true}
          path="/tag/:topicName"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/all"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/all/:topicName"
        />
        <Route
          component={[Function]}
          path="/all/p/:postId"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/c/:slug"
        />
        <Route
          component={[Function]}
          path="/c/:slug/members"
        />
        <Route
          component={[Function]}
          path="/c/:slug/m/:id"
        />
        <Route
          component={[Function]}
          path="/m/:id"
        />
        <Route
          component={[Function]}
          path="/c/:slug/p/:postId"
        />
        <Route
          component={[Function]}
          path="/c/:slug/topics"
        />
        <Route
          component={[Function]}
          path="/c/:slug/settings"
        />
        <Route
          component={[Function]}
          path="/c/:slug/:topicName/p/:postId"
        />
        <Route
          component={[Function]}
          path="/c/:slug/:topicName"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/n/:networkSlug"
        />
        <Route
          component={[Function]}
          path="/n/:networkSlug/p/:postId"
        />
        <Route
          component={[Function]}
          path="/n/:networkSlug/members"
        />
        <Route
          component={[Function]}
          path="/n/:networkSlug/m/:id"
        />
        <Route
          component={[Function]}
          path="/n/:networkSlug/settings"
        />
        <Route
          component={[Function]}
          path="/n/:networkSlug/communities"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/n/:networkSlug/:topicName"
        />
        <Route
          component={[Function]}
          path="/events"
        />
        <Route
          component={[Function]}
          path="/settings"
        />
        <Route
          component={[Function]}
          path="/search"
        />
        <Route
          component={[Function]}
          key="/signup/upload-photo"
          path="/signup/upload-photo"
        />
        <Route
          component={[Function]}
          key="/signup/add-location"
          path="/signup/add-location"
        />
        <Route
          component={[Function]}
          key="/signup/add-skills"
          path="/signup/add-skills"
        />
        <Route
          component={[Function]}
          key="/signup/review"
          path="/signup/review"
        />
        <Route
          component={[Function]}
          key="/create-community/name/:networkId"
          path="/create-community/name/:networkId"
        />
        <Route
          component={[Function]}
          key="/create-community/name"
          path="/create-community/name"
        />
        <Route
          component={[Function]}
          key="/create-community/domain"
          path="/create-community/domain"
        />
        <Route
          component={[Function]}
          key="/create-community/review"
          path="/create-community/review"
        />
      </Switch>
    </div>
    <div
      data-stylename="sidebar"
    >
      <Route
        component={[Function]}
        exact={true}
        path="/c/:slug/p/new"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/c/:slug"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/c/:slug/:topicName/p/new"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/c/:slug/:topicName"
      />
      <Route
        component={[Function]}
        path="/c/:slug/m/:id"
      />
      <Route
        component={[Function]}
        path="/n/:slug/m/:id"
      />
      <Route
        component={[Function]}
        path="/m/:id"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/n/:slug"
      />
    </div>
    <div
      data-stylename="detail hidden"
      id="detail-column"
    >
      <Route
        component={[Function]}
        key="/events/:eventId"
        path="/events/:eventId"
      />
      <Route
        component={[Function]}
        key="/all/p/:postId(\\\\d+)"
        path="/all/p/:postId(\\\\d+)"
      />
      <Route
        component={[Function]}
        key="/c/:slug/p/:postId(\\\\d+)"
        path="/c/:slug/p/:postId(\\\\d+)"
      />
      <Route
        component={[Function]}
        key="/n/:networkSlug/p/:postId(\\\\d+)"
        path="/n/:networkSlug/p/:postId(\\\\d+)"
      />
      <Route
        component={[Function]}
        key="/c/:slug/m/:id/p/:postId(\\\\d+)"
        path="/c/:slug/m/:id/p/:postId(\\\\d+)"
      />
      <Route
        component={[Function]}
        key="/c/:slug/:topicName/p/:postId(\\\\d+)"
        path="/c/:slug/:topicName/p/:postId(\\\\d+)"
      />
      <Route
        component={[Function]}
        key="/n/:slug/m/:id/p/:postId(\\\\d+)"
        path="/n/:slug/m/:id/p/:postId(\\\\d+)"
      />
    </div>
  </div>
  <Route
    component={[Function]}
    path="/t"
  />
  <Connect(SocketListener)
    location={
      Object {
        "pathname": "path",
      }
    }
  />
  <Connect(SocketSubscriber)
    type="community"
  />
  <Intercom />
  <Route
    exact={true}
    key="/all/p/new"
    path="/all/p/new"
  />
  <Route
    exact={true}
    key="/c/:slug/p/new"
    path="/c/:slug/p/new"
  />
  <Route
    exact={true}
    key="/c/:slug/:topicName/p/new"
    path="/c/:slug/:topicName/p/new"
  />
  <Route
    exact={true}
    key="/all/p/:postId/edit"
    path="/all/p/:postId/edit"
  />
  <Route
    exact={true}
    key="/c/:slug/p/:postId/edit"
    path="/c/:slug/p/:postId/edit"
  />
  <Route
    exact={true}
    key="/c/:slug/m/:id/p/:postId/edit"
    path="/c/:slug/m/:id/p/:postId/edit"
  />
  <Route
    exact={true}
    key="/c/:slug/:topicName/p/:postId/edit"
    path="/c/:slug/:topicName/p/:postId/edit"
  />
  <Route
    exact={true}
    key="/n/:slug/m/:id/p/:postId/edit"
    path="/n/:slug/m/:id/p/:postId/edit"
  />
</div>
`;

exports[`shows NotFound if a currentUser is loaded and the community does not exist 1`] = `<withRouter(Connect(NotFound)) />`;
